src/metrics.rs

```diff
@@ -1,132 +1,16 @@
-use futures01::Future;
-use hotmic::{
-    snapshot::{Snapshot, TypedMeasurement},
-    Controller, Receiver,
-};
-use hyper::{
-    server::conn::AddrStream,
-    service::{make_service_fn, service_fn_ok},
-    Body, Request, Response, Server,
-};
-use std::net::SocketAddr;
-use tracing::field;
+use metrics_runtime::{Controller, Receiver};
+use once_cell::sync::OnceCell;
 
-pub type Sink = hotmic::Sink<&'static str>;
+pub static CONTROLLER: OnceCell<Controller> = OnceCell::new();
 
-/// Build the metrics receiver, controller and sink
-pub fn build() -> (Controller, Sink) {
-    let mut receiver = Receiver::builder().build();
-    let controller = receiver.get_controller();
-    let sink = receiver.get_sink();
+pub fn init() {
+    let receiver = Receiver::builder()
+        .build()
+        .expect("failed to create receiver");
 
-    std::thread::spawn(move || {
-        receiver.run();
-    });
+    if CONTROLLER.set(receiver.controller()).is_err() {
+        panic!("failed to set receiver. metrics system already initialized.")
```

Comment:
It might be useful to have this return a result so it can be called multiple times, eg in a bench suite.